homeworks/src/main/java/hw40/dao/OrderRepository.java

```diff
@@ -0,0 +1,10 @@
+package hw40.dao;
+
+import hw40.entity.Order;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+
+public interface OrderRepository extends JpaRepository<Order, Long> {
+    @Query("SELECT o FROM Order o JOIN FETCH o.orderProducts op JOIN FETCH op.product p WHERE o.id = ?1")
```

Comment:
What if an order exists, but products were not assigned to it? Wouldn't it be better to do it via left join? Don't forget, that the left join might return duplicates, so you might need to use distinct as well.